src/promptflow/promptflow/_sdk/_service/utils/utils.py

```diff
@@ -39,25 +40,40 @@ def get_port_from_config(create_if_not_exists=False):
     (HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE).touch(mode=read_write_by_user(), exist_ok=True)
     with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "r", encoding=DEFAULT_ENCODING) as f:
         service_config = load_yaml(f) or {}
-        port = service_config.get("service", {}).get("port", None)
+        port = service_config.get(sys.executable, {}).get("port", None)
     if not port and create_if_not_exists:
         with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "w", encoding=DEFAULT_ENCODING) as f:
```

Comment:
do we need to add any protect to this file when writing from multiple process?

Reply:
added